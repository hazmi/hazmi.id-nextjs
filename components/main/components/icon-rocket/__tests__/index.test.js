import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import IconRocket from '../index'

describe('With Enzyme', () => {
  it('Show the correct default svg', () => {
    const wrapper = shallow(<IconRocket />)
    expect(wrapper.find('path').prop('fill')).toEqual('#444')
  })
  it('Show the correct custom color', () => {
    const wrapper = shallow(<IconRocket fill="red" />)
    expect(wrapper.find('path').prop('fill')).toEqual('red')
  })
})

describe('With Snapshot Testing', () => {
  it('Show the correct default svg', () => {
    const component = renderer.create(<IconRocket />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Show the correct custom color', () => {
    const component = renderer.create(<IconRocket fill="red" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})