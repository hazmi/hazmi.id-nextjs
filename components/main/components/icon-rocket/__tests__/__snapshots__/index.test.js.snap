// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Show the correct custom color 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M7 8v3.5l-4 3V20h4a2 2 0 0 0 2 2h2v1h2v-1h2a2 2 0 0 0 2-2h4v-5.5l-4-3V8c0-3.156-1.914-7-5-7-3.086 0-5 3.844-5 7zm6 12h2V8c0-2.193-1.398-5-3-5S9 5.807 9 8v12h2v-6h2v6zm-8-4.5L7 14v4H5v-2.5zM17 18v-4l2 1.5V18h-2zm-5-7a2 2 0 1 1 0-4 2 2 0 0 1 0 4z"
    fill="red"
    fillRule="evenodd"
  />
</svg>
`;

exports[`With Snapshot Testing Show the correct default svg 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M7 8v3.5l-4 3V20h4a2 2 0 0 0 2 2h2v1h2v-1h2a2 2 0 0 0 2-2h4v-5.5l-4-3V8c0-3.156-1.914-7-5-7-3.086 0-5 3.844-5 7zm6 12h2V8c0-2.193-1.398-5-3-5S9 5.807 9 8v12h2v-6h2v6zm-8-4.5L7 14v4H5v-2.5zM17 18v-4l2 1.5V18h-2zm-5-7a2 2 0 1 1 0-4 2 2 0 0 1 0 4z"
    fill="#444"
    fillRule="evenodd"
  />
</svg>
`;
